saltswap/record.py

```diff
@@ -0,0 +1,176 @@
+"""
+File to record saltswap simulation data with netcdf.
+"""
+from netCDF4 import Dataset
+import numpy as np
+import simtk
+
+# The simulation parameters that will be recorded from Swapper.
+swapper_control_attributes = ('temperature', 'pressure', 'delta_chem', 'npert')
+
+# Establishing default units for recording data. The netcdf file will also be storing units where appropriate.
+variable_states = {'volume':simtk.unit.nanometer**3, 'potential energy':simtk.unit.kilojoule_per_mole}
+
+# All variables that will be stored in the sampler state group
+sample_variables = ['identities','species counts','cumulative work','proposal','naccepted']
+for var in variable_states:
+    sample_variables.append(var)
+
+class CreateNetCDF(object):
+    """
+    Class to record data from MD and saltswap simulations
+    """
+    def __init__(self, filename, zlib=True):
+        """
+        Create a netcdf file to store saltswap simulation data.
+
+        Parameters
+        ----------
+        filename: str
+            the name of the netcdf file that will be created
+        """
+        self.ncfile = Dataset(filename, 'w', format='NETCDF4', zlib=zlib)
```

Comment:
I'm surprised that `Dataset` has a `zlib` option, since the API doesn't document this:
http://unidata.github.io/netcdf4-python/#netCDF4.Dataset

I thought you had to use the `zlib` argument individually for variables you wanted to compress:
http://unidata.github.io/netcdf4-python/#netCDF4.Dataset.createVariable